susemanager-utils/testing/docker/scripts/schema_idempotency_test_pgsql.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/python
+import argparse
+import difflib
+import distutils.version as DV
+import json
+import os
+import re
+import shutil
+import subprocess
+import sys
+
+
+def file_exists(path):
+    try:
+        open(path, 'r')
+        return(True)
+    except FileNotFoundError:
+        return(False)
+
+
+def directory_exists(path):
+    if os.path.isdir(path):
+        return(True)
+    return(False)
+
+
+def get_all_files_rpm_pr(prFile):
+    with open(prFile, 'r') as f:
+        changedFiles = []
+        files = json.loads(read(f))['files']
+        for file in filesDict:
+            if file['status'] == 'modified' or file['status'] == 'added':
+                changedFiles.append(file['filename'])
+    return(changedFiles)
+
+
+def get_all_files_since(version, schemaPath):
+    files = []
+    for element in os.listdir(schemaPath):
+        elementVer = re.search(
+            '^susemanager-schema-(\d+\.\d+[\d|.]*)-to-susemanager-schema-\d+\.\d+[\d|.]*$', element)
+        if elementVer:
+            # StrictVersion does not work as we have some migrations with version X.Y.W.Z
+            # https://hg.python.org/cpython/file/tip/Lib/distutils/version.py#l93
+            if DV.LooseVersion(elementVer.group(1)) >= DV.LooseVersion(version):
+                basePath = schemaPath + '/' + element
+                for sqlFile in os.listdir(basePath):
+                    if re.search('^.*\.(sql|sql\.postgresql|sql\.oracle)$', sqlFile):
+                        files.append(basePath + '/' + sqlFile)
+    return(files)
+
+
+def find_latest_version(schemaPath):
+    latestVersion = '0.0'
+    for element in os.listdir(schemaPath):
+        elementVer = re.search(
+            '^susemanager-schema-\d+\.\d+[\d|.]*-to-susemanager-schema-(\d+\.\d+[\d|.]*)$', element)
+        if elementVer:
+            if DV.LooseVersion(elementVer.group(1)) > DV.LooseVersion(latestVersion):
+                latestVersion = elementVer.group(1)
+    return(latestVersion)
+
+
+def find_next_version(schemaPath):
+    lastVersion = find_latest_version(schemaPath)
+    return(lastVersion.rsplit('.', 1)[0] + '.' + str(int(lastVersion.rsplit('.', 1)[1])+1))
+
+
+def create_fake_migration_path(schemaPath, newVersion, prFile=None, version=None):
+    if prFile:
+        print("Creating migration path with the scripts from the PR")
+        files = get_all_files_from_pr(prFile, schemaPath)
+    if version:
+        print("Creating migration path with all scripts since %s" % version)
+        files = get_all_files_since(version, schemaPath)
+    lastVersion = find_latest_version(schemaPath)
+    fakePath = schemaPath + \
+        '/susemanager-schema-%s-to-susemanager-schema-%s/' % (
+            lastVersion, newVersion)
+    print("Creating: " + fakePath)
+    os.mkdir(fakePath)
+    for migrationFile in files:
+        shutil.copy(migrationFile, fakePath)
+    #run_command(sys.path[0] + "/build-schema.sh")
```

Comment:
if the command is commented out i would not commit it

Reply:
Ooops. This was part of a test. Good catch.

Removed.